KOGITO-6883: [SWF Editor] Create a Language Service for Serverless Workflow

tiagobento opened the pull request:
Jira: https://issues.redhat.com/browse/KOGITO-6883

---

On this PR:
- Change `syncpack` to allow for a different version of Monaco to be used on SWF Editor
- Created the new `serverless-workflow-language-service` package.
- Added a new flag to allow for changing the content of an editor without showing the Loading overlay
- Some renames to better represent Diagram VS. Text editors, instead of SVG/Mermaid VS. Monaco.
- Moved all augmentation logic from the SWF Text Editor to the new Language Service package.
- In VS Code, now the default is to open the SWF Editor file using VS Code's built-in text editor. The SWF Diagram Editor is shown side-by-side with the built-in text editor (There's a new configuration option to change that). This behavior was based on the Markdown support for VS Code.
- Changed the `@kie-tools-core/vscode-extension` package to allow for using the CustomTextEditor API as well, where the Multiplying Architecture Editor is synced with the TextDocument on VS Code's backend. This makes multi-tab editing very pleasant and allows for better native support of text editing tools provided by the VS Code platform.
- Added keyboard shortcuts to switch between the built-in text editor in VS Code and our SWF Diagram Editor.

tiagobento commented:
@tomasdavidorg Can you please test this?

tomasdavidorg commented:
Hi @tiagobento,
I build the SWF VsCode extension and after I tried to open the diagram panel I got: 

`command 'extension.kogito.swf.openAsDiagram' not found`

![image](https://user-images.githubusercontent.com/4498639/166926289-79c73839-a76c-497c-9a35-6b314cc9c4ed.png)

tomasdavidorg commented on packages/vscode-extension-serverless-workflow-editor/src/extension/ServerlessWorkflowEditorChannelApiProducer.ts:
```diff
@@ -54,54 +55,16 @@ export class ServerlessWorkflowEditorChannelApiProducer implements KogitoEditorC
-    });
-
-    // TODO: This is a workaround
+    // TODO: This casting to `unknown` first is a workaround
```
Another TODO

tomasdavidorg commented on packages/vscode-extension-serverless-workflow-editor/src/extension/languageService/VsCodeSwfLanguageService.ts:
```diff
@@ -0,0 +1,108 @@
+      },
+      config: {
+        shouldDisplayRhhccIntegration: async () => {
+          // FIXME: Tiago: This should take the OS into account as well. RHHCC integration only works on macOS.
```
Another FIXME

tiagobento commented:
@tomasdavidorg Thanks for the review!!

I just added the Jiras to the TODOs/FIXMEs and fixed the issue you found.

Now you should be able to use the extension normally even without the Red Hat Authentication extension installed.

tomasdavidorg reviewed (approved):
Thanks @tiagobento. Looks good to me now.
